packages/medusa/src/api-v2/admin/uploads/middlewares.ts

```diff
@@ -1,25 +1,36 @@
 import multer from "multer"
 import { MiddlewareRoute } from "../../../loaders/helpers/routing/types"
 import { authenticate } from "../../../utils/authenticate-middleware"
+import { validateAndTransformQuery } from "../../utils/validate-query"
+import { retrieveUploadConfig } from "./query-config"
+import { AdminGetUploadParams } from "./validators"
 
-const upload = multer({ dest: "uploads/" })
+// TODO: For now we keep the files in memory, as that's how they get passed to the workflows
+// This will need revisiting once we are closer to prod-ready v2, since with workflows and potentially
+// services on other machines using streams is not as simple as it used to be.
+const upload = multer({ storage: multer.memoryStorage() })
```

Comment:
question/thought: I guess we will need to somehow integrate this part of the flow with the file provider so that we can store the uploads on a remote server e.g. S3 if you are using the S3 plugin. Do you already have thoughts on this?

Reply:
@olivermrbl So what happens currently is, the entire data/file the user sent will be kept in memory, passed through the workflow, to the file provider, and then it gets converted from in-memory binary content to a buffer and stored as a file.

If we decide that, for uploads, we want to avoid using workflows and call a module directly then we can probably support streaming more easily, as you wouldn't need any sort of serialization. That might be the way to go, but honestly for now it doesn't matter as much, as supporting that won't be a major change.